packages/engine/vscode/src/lib/filemanager.ts

```diff
@@ -1,91 +1,100 @@
+    const absPath = absolutePath(path)
+    const uri = Uri.file(absPath)
+    return commands.executeCommand('vscode.open', uri)
   }
   /** Set the content of a specific file */
   async writeFile(path: string, data: string): Promise<void> {
-    const absPath = absolutePath(path);
-    const uri = Uri.file(absPath);
-    const encoder = new TextEncoder();
-    const uint8Array = encoder.encode(data);
-    return workspace.fs.writeFile(uri, Uint8Array.from(uint8Array));
+    const absPath = absolutePath(path)
+    const uri = Uri.file(absPath)
+    const encoder = new TextEncoder()
+    const uint8Array = encoder.encode(data)
+    return workspace.fs.writeFile(uri, Uint8Array.from(uint8Array))
   }
   /** Return the content of a specific file */
   async readFile(path: string): Promise<string> {
-    const absPath = absolutePath(path);
-    const uri = Uri.file(absPath);
-    return workspace.fs.readFile(uri).then(content => Buffer.from(content).toString("utf-8"));
+    const absPath = absolutePath(path)
+    const uri = Uri.file(absPath)
+    return workspace.fs.readFile(uri).then(content => Buffer.from(content).toString("utf-8"))
   }
   /** Remove a file */
   async remove(path: string): Promise<void> {
-    const absPath = absolutePath(path);
-    const uri = Uri.file(absPath);
+    const absPath = absolutePath(path)
+    const uri = Uri.file(absPath)
     return workspace.fs.delete(uri)
   }
   /** Change the path of a file */
   async rename(oldPath: string, newPath: string): Promise<void> {
-    const source = Uri.file(absolutePath(oldPath));
-    const target = Uri.file(absolutePath(newPath));
-    return workspace.fs.rename(source, target);
+    const source = Uri.file(absolutePath(oldPath))
+    const target = Uri.file(absolutePath(newPath))
+    return workspace.fs.rename(source, target)
   }
   /** Upsert a file with the content of the source file */
   async copyFile(src: string, dest: string): Promise<void> {
-    const source = Uri.file(absolutePath(src));
-    const target = Uri.file(absolutePath(dest));
-    return workspace.fs.copy(source, target);
+    const source = Uri.file(absolutePath(src))
+    const target = Uri.file(absolutePath(dest))
+    return workspace.fs.copy(source, target)
   }
   /** Create a directory */
   async mkdir(path: string): Promise<void> {
-    const uri = Uri.file(absolutePath(path));
-    return workspace.fs.createDirectory(uri);
+    const uri = Uri.file(absolutePath(path))
+    return workspace.fs.createDirectory(uri)
   }
   /** Get the list of files in the directory */
   async readdir(path: string): Promise<string[]> {
-    const absPath = absolutePath(path);
-    const uri = Uri.file(absPath);
-    return workspace.fs.readDirectory(uri).then(data => data.map(([path]) => path));
+    const absPath = absolutePath(path)
+    const uri = Uri.file(absPath)
+    return workspace.fs.readDirectory(uri).then(data => data.map(([path]) => relativePath(path)))
```

Comment:
I was about to ask, is that fine to do this : `data => data.map(([path]) => relativePath(path))` 
@0mkara @bunsenstraat

Reply:
I haven't checked that particular condition. Can't say without testing. Will do a test and let you know here.